src/lib/util.ts

```diff
@@ -3,9 +3,9 @@ import { BigNumber } from "bignumber.js";
 
 export default class Util {
 
-  public static fromWei(amount: BigNumber): number {
+  public static fromWei(amount: BigNumber): BigNumber {
     const web3 = Arc.Utils.getWeb3();
-    return Number(web3.fromWei(amount, "ether"));
+    return web3.fromWei(amount, "ether");
```

Comment:
I am pretty sure that `web3.fromWei(amount, "ether")` returns `BigNumber|string`, hence I think you should do:

`return web3.toBigNumber(web3.fromWei(amount, "ether"));`

Reply:
What about `toWei`?  I think it should return BigNumber too.

Reply:
`web3.fromWei` only returns a string if you give it a `number | string`, we are giving it a `BigNumber`.

about the type of `toWei`, see my previous comment, added a todo for future refactoring.